avocado/utils/configure_network.py

```diff
@@ -10,193 +10,323 @@
+        try:
+            return _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to get hw address: {}".format(ex))
+
+    def get_link_state(self):
+        """Method used to get the link state of an interface.
 
-    def __init__(self, host, port=None, peer_user=None,
-                 key=None, peer_password=None):
+        This method will return 'up', 'down' or 'unknown', based on the network
+        interface state. Or it will raise a NWException if is unabble to get
+        the interface state.
         """
-        create a object for accesses remote machine
+        cmd = "cat /sys/class/net/{}/operstate".format(self.name)
+        try:
+            return _run_command(cmd, self.remote_session)
+        except process.CmdError as e:
+            msg = ('Failed to get link state. Maybe the interface is '
+                   'missing. {}'.format(e))
+            raise NWException(msg)
+
+    def get_mtu(self):
+        pass
+
+    def ping_check(self, peer_ip, options=None):
+        """This method will try to ping a peer address (IPv4 or IPv6).
+
+        You should provide a IPv4 or IPV6 that would like to ping. This
+        method will try to ping the peer and if fails it will raise a
+        NWException.
+
+        :param peer_ip: Peer IP address (IPv4 or IPv6)
+        :param options: ping command options. Default is None
         """
+        cmd = "ping -I {} {}".format(self.name, peer_ip)
+        if options is not None:
+            cmd = "{} {}".format(cmd, options)
         try:
-            self.session = Session(host, port=port, user=peer_user,
-                                   key=key, password=peer_password)
-        except Exception as ex:  # pylint: disable=W0703
-            log.error("connection not established to peer machine: %s", ex)
+            _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to ping: {}".format(ex))
+
+    def save(self, ipaddr, netmask):
+        """
+        Utility assign a IP  address (given to this utility ) to  Interface
+        And generate interface file in sysfs based on distribution
+
+        :param ipaddr : ip address which need to configure for interface
+        :param netmask: Netmask which is associated  to provided IP
+        :param interface_type: Interface type IPV4 or IPV6 , default is
+                               IPV4 style
+        """
+        current_distro = distro.detect()
 
-    def set_mtu_peer(self, peer_interface, mtu):
+        filename = "ifcfg-{}".format(self.name)
+        if current_distro.name in ['rhel', 'fedora']:
+            path = "/etc/sysconfig/network-scripts"
+        elif current_distro.name == 'SuSE':
+            path = "/etc/sysconfig/network"
+        else:
+            msg = 'Distro not supported by API. Could not save ipaddr.'
+            raise NWException(msg)
+
+        if ipaddr not in self.get_ipaddrs():
```

Comment:
maybe you want to do this check at the beginning of the method, before probing the distro?

Reply:
good catch.